s3/src/test/scala/akka/stream/alpakka/s3/scaladsl/S3IntegrationSpec.scala

```diff
@@ -1040,75 +1045,59 @@ trait S3IntegrationSpec
- * AWS_ACCESS_KEY_ID=TESTKEY AWS_SECRET_ACCESS_KEY=TESTSECRET aws --endpoint-url http://localhost:9000 s3api create-bucket --bucket my-test-us-east-1
- * AWS_ACCESS_KEY_ID=TESTKEY AWS_SECRET_ACCESS_KEY=TESTSECRET aws --endpoint-url http://localhost:9000 s3api create-bucket --bucket my.test.frankfurt
-
- * NOTE: Ideally you would run the following commands to create a versioned bucket in Minio however its not working, see https://github.com/akka/alpakka/issues/2750
- * AWS_ACCESS_KEY_ID=TESTKEY AWS_SECRET_ACCESS_KEY=TESTSECRET aws --endpoint-url http://localhost:9000 s3api create-bucket --bucket my-bucket-with-versioning
- * AWS_ACCESS_KEY_ID=TESTKEY AWS_SECRET_ACCESS_KEY=TESTSECRET aws --endpoint-url http://localhost:9000 s3api put-bucket-versioning --bucket my-bucket-with-versioning --versioning-configuration Status=Enabled
- *
- * Run the tests from inside sbt:
- * s3/testOnly *.MinioS3IntegrationSpec
+ * For this test, you need a have docker installed and running
  */
-class MinioS3IntegrationSpec extends S3IntegrationSpec {
-  import MinioS3IntegrationSpec._
+
+class MinioS3IntegrationSpec
+    extends TestKit(ActorSystem("MinioS3IntegrationSpec"))
+    with S3IntegrationSpec
+    with MinioS3Test {
 
   override val defaultRegionContentCount = 0
   override val otherRegionContentCount = 0
 
-  override def config() =
-    ConfigFactory.parseString(s"""
-                                 |alpakka.s3 {
-                                 |  aws {
-                                 |    credentials {
-                                 |      provider = static
-                                 |      access-key-id = $accessKey
-                                 |      secret-access-key = $secret
-                                 |    }
-                                 |  }
-                                 |  endpoint-url = "$endpointUrlVirtualHostStyle"
-                                 |}
-    """.stripMargin).withFallback(super.config())
-
-  override def otherRegionSettingsPathStyleAccess =
-    S3Settings()
-      .withCredentialsProvider(StaticCredentialsProvider.create(AwsBasicCredentials.create(accessKey, secret)))
-      .withEndpointUrl(endpointUrlPathStyle)
-      .withAccessStyle(PathAccessStyle)
-
-  override def invalidCredentials: S3Settings =
-    S3Settings()
-      .withCredentialsProvider(
-        StaticCredentialsProvider.create(AwsBasicCredentials.create("invalid", "invalid"))
-      )
-
-  override def attributes: Attributes = attributes(S3Settings()(actorSystem))
-
-  override def attributes(s3Settings: S3Settings): Attributes = {
-    S3Attributes.settings(
-      s3Settings
-        .withForwardProxy(
-          ForwardProxy.http("localhost", 9000)
-        )
+  override lazy val defaultS3Settings: S3Settings = s3Settings
+    .withS3RegionProvider(
+      new AwsRegionProvider {
+        val getRegion: Region = Region.US_EAST_1
+      }
     )
+    .withAccessStyle(PathAccessStyle)
+
+  override def beforeAll(): Unit = {
```

Comment:
This is a replacement for what was previously in the `docker-compose.yml` `minio_prep` section. We can just programatically create the S3 buckets needed for the tests in the standard scalatest `beforeAll`